src/main/java/interaction/rule/TankBulletInteractionRule.java

```diff
@@ -37,31 +44,31 @@ public TankBulletInteractionRule(GameObjectManager gameObjectManager) {
             actions.add(new DeleteAction(bullet, gameObjectManager));
 
             return actions;
-        } catch (InteractionRuleException e) {
+        } catch (InteractionRuleException | WrongObjectIdException e) {
             e.printStackTrace();
         }
         return Collections.emptySet();
     }
 
-    private Tank getTank(GameObject first, GameObject second) throws InteractionRuleException {
+    private Tank getTank(GameObject first, GameObject second) throws InteractionRuleException, WrongObjectIdException {
         if (first.getTypeId() == Type.TANK) {
-            return (Tank) first;
+            return tankObjectManager.getObjectById(first.getId());
         }
 
         if (second.getTypeId() == Type.TANK) {
-            return (Tank) second;
+            return tankObjectManager.getObjectById(second.getId());
         }
 
         throw new InteractionRuleException();
     }
 
-    private Bullet getBullet(GameObject first, GameObject second) throws InteractionRuleException {
+    private Bullet getBullet(GameObject first, GameObject second) throws InteractionRuleException, WrongObjectIdException {
         if (first.getTypeId() == Type.BULLET) {
-            return (Bullet) first;
+            return bulletObjectManager.getObjectById(first.getId());
         }
 
         if (second.getTypeId() == Type.BULLET) {
```

Comment:
if first object is not bullet then second is definitely bullet. We have such contract when rule is choosen

Reply:
doesn't it increase readability of the code? no need to know that such contract exists, also is it good when code rely on that?

Reply:
@ramazan-bagaev the code you've written seems procedural. In our case we try to write in object oriented paradigm. 

>no need to know that such contract exists

It seems to me that its better to comment such contract:

```
/**
* defines what actions to take when two game objects interact with each other.
* When this interface is implemented for Interaction of T and V types then first object is always
* of type T and second is V
*/
public interface InteractionRule {
...
```

in InteractionRule interface so then its not necessary to implement such if statements in every implementation and produce a lot of risks due to consensus breakage which leads to throw new InteractionRuleException();